src/GitVersion.Core/VersionCalculation/MainlineVersionCalculator.cs

```diff
@@ -86,9 +86,22 @@ public SemanticVersion FindMainlineModeVersion(BaseVersion baseVersion)
 
     public SemanticVersionBuildMetaData CreateVersionBuildMetaData(ICommit? baseVersionSource)
     {
-        var commitLog = this.repositoryStore.GetCommitLog(baseVersionSource, context.CurrentCommit);
-        var commitsSinceTag = commitLog.Count();
-        this.log.Info($"{commitsSinceTag} commits found between {baseVersionSource} and {context.CurrentCommit}");
+        int commitsSinceTag = 0;
+        if (context.CurrentCommit != null)
+        {
+            var commitLogs = this.repositoryStore.GetCommitLog(baseVersionSource, context.CurrentCommit);
+
+            var ignore = context.FullConfiguration.Ignore;
+            if (!ignore.IsEmpty)
+            {
+                var hashSetLazy = new Lazy<HashSet<string>>(() => new(ignore.ShAs));
```

Comment:
Why is this `HashSet` wrapped in a `Lazy`? Can you please explain it with a comment?

Reply:
I thought because of performance reason. This do not need to be lazy. I have changed it